Sources/VLCTime.m

```diff
@@ -24,7 +25,18 @@
 
 #import "VLCTime.h"
 
+@interface VLCTime(Private)
+- (void)initDateComponents;
+@end
+
 @implementation VLCTime
+static NSDateComponents *components = nil;
+
+- (void)initDateComponents
+{
+    if (!components)
+        components = [[NSDateComponents alloc] init];
```

Comment:
turns out I've been mixing up the expensiveness of NSDateFormatter with NSDateComponents.
You can use this one actually directly.

Reply:
👍